### PR TITLE
Expand list of test cases to run with those from the base test run

### PR DESCRIPTION
**Pull request by alexivanov**
This PR updates the `runAllTests` logic to expand the list of test cases to include those from the base test run. This is necessary to prevent a drop in coverage when the project config test cases are updated. Such updates have the potential to entirely replace the set of sessions to run and will occur more frequently in the future.

### CODE REVIEW

File: packages/replay-orchestrator/src/test-run/parallel-tests/run-all-tests.ts [hunk 1]
```diff
@@ -402,3 +418,18 @@ const getTestCasesWithBaseTestRunId = async ({
...
 };
+
+const shouldUseBaseTestRunTestCases = async ({
+  client,
+  baseTestRunId,
+}: {
+  client: AxiosInstance;
+  baseTestRunId: string;
+}) => {
+  const testRun = await getTestRun({ client, testRunId: baseTestRunId });
```
Comment (Que3216) [inline packages/replay-orchestrator/src/test-run/parallel-tests/run-all-tests.ts:429]: Haven't we already loaded this, stored as fallbackTestRun?

**Review by Que3216 — APPROVED** [review]
Looks good, though I don't know if we need the extra call

File: packages/replay-orchestrator/src/test-run/parallel-tests/run-all-tests.ts (hunk 1 above)
Comment (alexivanov) [reply packages/replay-orchestrator/src/test-run/parallel-tests/run-all-tests.ts]: 🤦 Yep, https://github.com/alwaysmeticulous/meticulous-sdk/pull/340/commits/8c90537f00c991b576784068958c6314190cc0e6 drops the extra call

File: packages/replay-orchestrator/src/test-run/parallel-tests/run-all-tests.ts [hunk 2]
```diff
@@ -402,3 +418,18 @@ const getTestCasesWithBaseTestRunId = async ({
...
+
+  // We only want to include the test cases from the base test run if it has no base test run itself, i.e the
+  // the base test run is for a "push" event.
+  // This safeguards against the test run expanding indefinitely.
```
[the remark this answers is not in the document]
Comment (Que3216) [reply packages/replay-orchestrator/src/test-run/parallel-tests/run-all-tests.ts:427]: (i.e. we could store `fallbackTestRun?.configData.sessionIdsExecutedForBackCompatOnly`, and then filter these ones out)

**Pull request merged**